Adafruit_FXOS8700.h

```diff
@@ -92,10 +186,28 @@ typedef enum {
 /*=========================================================================*/
 
 /*=========================================================================
-    RAW GYROSCOPE DATA TYPE
+    OPTIONAL MAGNETOMETER OVERSAMPLING SETTINGS
+    -----------------------------------------------------------------------*/
+/*!
+    Range settings for the accelerometer sensor.
+*/
+typedef enum {
+  MAG_OSR_0 = 0x00, /**< Mag oversampling ratio = 0 */
+  MAG_OSR_1 = 0x04, /**< Mag oversampling ratio = 1 */
+  MAG_OSR_2 = 0x08, /**< Mag oversampling ratio = 2 */
+  MAG_OSR_3 = 0x0C, /**< Mag oversampling ratio = 3 */
+  MAG_OSR_4 = 0x10, /**< Mag oversampling ratio = 4 */
+  MAG_OSR_5 = 0x14, /**< Mag oversampling ratio = 5 */
+  MAG_OSR_6 = 0x18, /**< Mag oversampling ratio = 6 */
+  MAG_OSR_7 = 0x1C, /**< Mag oversampling ratio = 7 */
+} fxos8700MagOSR_t;
```

Comment:
This isn't needed since you're using RegisterBits. With that, it's just a simple write of the value `0` to `7` since you access those 3 bits directly.

Reply:
Nice. Changed, thanks.

Does same go for lines `89-93` & `100-104` *(Adafruit_FXOS8700.h)* except `0` to `1` & `0` to `2`? It uses RegisterBits as well.
![image](https://user-images.githubusercontent.com/32855321/133831112-9edc03c3-237c-4aa0-97d0-04d4f8e1279f.png)
![image](https://user-images.githubusercontent.com/32855321/133831144-ab9c2efe-2439-4c6a-8a35-68c599b79e25.png)

Reply:
Those look fine. Two different notations, hex and binary, but in general OK. It may be a little confusing with those, since in both cases the bits of interest have 0 offset in the registers.

![image](https://user-images.githubusercontent.com/8755041/133851960-97f0792c-2dab-47ca-936d-a46ba508b2a5.png)

![image](https://user-images.githubusercontent.com/8755041/133852066-0e3b6e76-d7ff-474d-927c-1f62ed6251a0.png)

What RegisterBits does is more interesting when there is an offset, like `2` in the case of the OSR bits:
![image](https://user-images.githubusercontent.com/8755041/133851836-061d041e-dd8a-49cd-84d9-20f0d56741e8.png)
That offset is one of the parameters passed in when creating the instance. So it takes care of the shifting for you under the hood. Your code only has to think in terms of a 3 bit value.